### ISSUE TITLE
Large Softmax hangs through compiler

### ISSUE THREAD

**Issue by ppadjinTT** [tenstorrent/tt-xla#2878]
### Describe the bug

A softmax with large input tensors hangs in the compiler - no error, no output, just dangling python process. 

### Steps to reproduce the issue

On fresh tt-xla run this script 

```
import torch
import torch.nn.functional as F
import torch_xla
import torch_xla.core.xla_model as xm
import torch_xla.runtime as xr


def test_softmax_sdxl_attention():
    xr.set_device_type("TT")
    class Softmax(torch.nn.Module):
        def forward(self, x):
            return F.softmax(x, dim=3)

    model = Softmax()
    input_tensor = torch.randn(2, 10, 4096, 4096, dtype=torch.float32)

    model = model.to(xm.xla_device())
    input_tensor = input_tensor.to(xm.xla_device())

    model = torch.compile(model, backend="tt")

    with torch.no_grad():
        output = model(input_tensor)

    print("Here is the output:")
    print(output.cpu())

if __name__ == "__main__":
    test_softmax_sdxl_attention()
```

However, if smaller input shape is used, e.g. `(2, 10, 2048, 2048)`, the output is generated normally.
